File: app/src/main/java/com/georgcantor/wallpaperapp/ui/util/DynamicHeightImageView.java
package com.georgcantor.wallpaperapp.ui.util;

import android.content.Context;
import android.util.AttributeSet;

public class DynamicHeightImageView extends android.support.v7.widget.AppCompatImageView {

    private float whRatio = 0;

    public DynamicHeightImageView(Context context) {
        super(context);
    }

    public DynamicHeightImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public void setRatio(float ratio) {
        whRatio = ratio;
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        if (whRatio != 0) {
            int width = getMeasuredWidth();
            int heigth = (int) (whRatio * width);
            setMeasuredDimension(width, heigth);
        }
    }
}

File: app/src/main/java/com/georgcantor/wallpaperapp/ui/util/DynamicHeightImageView.kt
package com.georgcantor.wallpaperapp.ui.util

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView

class DynamicHeightImageView : AppCompatImageView {

    private var whRatio = 0f

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    fun setRatio(ratio: Float) {
        whRatio = ratio
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        if (whRatio != 0f) {
            val width = measuredWidth
            val height = (whRatio * width).toInt()
            setMeasuredDimension(width, height)
        }
    }
}
